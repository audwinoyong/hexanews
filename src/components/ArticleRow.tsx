import * as React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'

import placeholder from '../assets/placeholder.png'

type Props = {
  title: string,
  description: string
}

const styles = {
  author: {
    marginTop: 28
  },
  card: {
    display: 'flex',
    boxShadow: '0px 0.5px 0px 0px rgba(0, 0, 0, 0.2)'
  },
  date: {
    marginTop: 2,
  },
  details: {
    display: 'flex',
    flex: 4,
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  container: {
    marginTop: 24,
  },
  cover: {
    width: 151,
    height: 151,
  },
  imageContainer: {
    flex: 1,
    justifyContent: 'flex-end',
    display: 'flex',
  },
  title: {
    marginBottom: 8
  },
}

const ArticleRow: React.StatelessComponent<Props> = ({
  title,
  description
}) => {
  return (
    <div style={styles.container}>
      <Card style={styles.card}>
        <div style={styles.details as any}>
          <CardContent style={styles.content}>
            <Typography
              variant='headline'
              style={styles.title}
            >
              {title}
            </Typography>
            <Typography component='p'>
              {description}
            </Typography>
            <Typography
              variant='body1'
              style={styles.author}
            >
              {'Hillary Clinton'}
            </Typography>
            <Typography
              variant='body1'
              color='textSecondary'
              style={styles.date}
            >
              {'22 August 2018'}
            </Typography>
          </CardContent>
        </div>
        <div style={styles.imageContainer}>
          <CardMedia
            style={styles.cover}
            image={placeholder}
          />
        </div>
      </Card>
    </div>
  )
}

export default ArticleRow