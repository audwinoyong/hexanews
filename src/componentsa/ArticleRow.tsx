import * as React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'

import placeholder from '../assets/placeholder.png'

type Props = {
  title: string,
  description: string
}

const styles = {
  card: {
    display: 'flex',
    boxShadow: '0px 0.5px 0px 0px rgba(0, 0, 0, 0.2)'
  },
  details: {
    display: 'flex',
    flex: 4,
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    width: 151,
    height: 151,
  },
}

const ArticleRow: React.StatelessComponent<Props> = ({
  title,
  description
}) => {
  return (
    <div style={{marginTop: 24}}>
      <Card style={styles.card as any}>
        <div style={styles.details as any}>
          <CardContent style={styles.content}>
            <Typography variant='headline' style={{marginBottom: 8}}>{title}</Typography>
            <Typography component='p'>
              {description}
            </Typography>
          </CardContent>
        </div>
        <div style={{ flex: 1, justifyContent: 'flex-end', display: 'flex'}}>
          <CardMedia
            style={styles.cover}
            image={placeholder}
            title='Live from space album cover'
          />
        </div>
      </Card>
    </div>
  )
}

export default ArticleRow