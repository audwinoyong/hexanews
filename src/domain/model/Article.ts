/**
 * Model for the article.
 */
export type Article = {
  title: string,
  description: string
  category?: string
  authorId?: string
}