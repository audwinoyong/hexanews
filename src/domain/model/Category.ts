/**
 * Model for category object in our application.
 */

export type Category = {
  _id?: string,
  title: string,
  description: string,
}