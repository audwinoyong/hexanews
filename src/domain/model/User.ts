/**
 * Model for the user.
 */
export type User = {
  _id: string,
  email: string,
  name: string,
  description: string,
  createdAt: string,
}