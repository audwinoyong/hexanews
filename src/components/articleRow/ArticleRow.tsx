/**
 * Display article row component.
 */

import * as React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import * as moment from 'moment'
import { Dispatch } from 'redux'
import { push } from 'connected-react-router'

import placeholder from '../../assets/placeholder.png'
import styles from './styles'
import { Article } from '../../domain/model/Article'

type Props = {
  article: Article,
  authorName: string,
  dispatch: Dispatch<any>,
}

const ArticleRow: React.StatelessComponent<Props> = ({
  article,
  authorName,
  dispatch,
}) => {
  const sanitizedDescription = article.description.replace(/<(?:.|\n)*?>/gm, '')
  const articleDescription = sanitizedDescription.length > 128 ? `${sanitizedDescription.substring(0, 128)}...` : sanitizedDescription
  return (
    <div
      style={styles.container}
      onClick={() => dispatch(push(`articleDetail/${article._id}`))}
    >
      <Card style={styles.card}>
        <div style={styles.details as any}>
          <CardContent style={styles.content}>
            <Typography
              variant='headline'
              style={styles.title}
            >
              {article.title}
            </Typography>
            <Typography component='p'>
              {articleDescription}
            </Typography>
            <Typography
              variant='body1'
              style={styles.author}
            >
              {authorName}
            </Typography>
            <Typography
              variant='body1'
              color='textSecondary'
              style={styles.date}
            >
              {moment(article.createdAt).format('D MMMM YYYY')}
            </Typography>
          </CardContent>
        </div>
        <div style={styles.imageContainer}>
          <CardMedia
            style={styles.cover}
            image={placeholder}
          />
        </div>
      </Card>
    </div>
  )
}

export default ArticleRow