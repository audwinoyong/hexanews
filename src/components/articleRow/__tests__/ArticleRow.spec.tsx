import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import ArticleRow from '../ArticleRow'
import { Article } from '../../../domain/model/Article'

Enzyme.configure({ adapter: new Adapter() })

const mockArticle: Article = {
  _id: '2',
  title: 'def',
  description: 'tester',
  category: 'tester',
  authorId: 'tester',
}

describe('Article Row', () => {
  it('renders Article row', () => {
    const component = renderer.create(
      <ArticleRow
        article={mockArticle}
        dispatch={jest.fn()}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})