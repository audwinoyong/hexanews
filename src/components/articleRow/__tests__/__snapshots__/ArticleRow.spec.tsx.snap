// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Row renders Article row 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginTop": 24,
    }
  }
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
    style={
      Object {
        "boxShadow": "0px 0.5px 0px 0px rgba(0, 0, 0, 0.2)",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 4,
          "flexDirection": "column",
        }
      }
    >
      <div
        className="MuiCardContent-root-29"
        style={
          Object {
            "flex": "1 0 auto",
          }
        }
      >
        <h2
          className="MuiTypography-root-30 MuiTypography-title-36"
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 8,
            }
          }
        >
          def
        </h2>
        <p
          className="MuiTypography-root-30 MuiTypography-body1-39"
        >
          tester
        </p>
        <p
          className="MuiTypography-root-30 MuiTypography-body1-39"
          style={
            Object {
              "marginTop": 28,
            }
          }
        >
          tim
        </p>
        <p
          className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-53"
          style={
            Object {
              "marginTop": 2,
            }
          }
        >
          18 September 2018
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        className="MuiCardMedia-root-55"
        src={undefined}
        style={
          Object {
            "height": 151,
            "width": 151,
          }
        }
      />
    </div>
  </div>
</div>
`;
